### PR TITLE
Feature/recording state persistant

### PR DESCRIPTION
**Pull request by prinsniels**
Added persistent recording state. Now keeps recording state over application boot

### CODE REVIEW

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/App.java [hunk 1]
```diff
@@ -16,19 +19,34 @@
...
+
+    /**
+     * Clear the existing database
+     * @param appcontext
+     */
     public static void resetDatabase(Context appcontext) {
+        // close the existing database connection
         dbhelper.close();
-
+        // TODO: why not: appcontext.deleteDatabase(Database.getDataPath(appcontext).getName());?
```
Comment (wowtor) [inline app/src/main/java/cellscanner/wowtor/github/com/cellscanner/App.java:38]: like

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java [hunk 2]
```diff
@@ -109,7 +108,7 @@ public void onCreate() {
         Runnable timer = new Runnable() {
             @Override
             public void run() {
-                if (running) {
+                if (inRecordingState(getApplicationContext())) {
```
Comment (wowtor) [inline app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java]: why this check? shouldn't the timer be stopped in the first place when not in recording state?

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import android.content.Context;
+import android.content.SharedPreferences;
+
+
+/**
+ * Responsible for controlling the recording functionality
+ */
+public class Recorder {
+    // TODO: PICK A NAME NOT `MY_PREFS_NAME`
+    final private static String PREFS_NAME = "MY_PREFS_NAME";
```
Comment (wowtor) [inline app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java]: Should be application name? `App.TITLE`?

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package cellscanner.wowtor.github.com.cellscanner.recorder;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+
+/**
+ * Responsible for controlling the recording functionality
+ */
+public class Recorder {
```
Comment (wowtor) [inline app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java:12]: It is not actually a recorder; maybe pick a name that fits its purpose, like `Preferences`?

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java (hunk 2 above)
Comment (prinsniels) [reply app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java]: Correct, but the timer check is not in this pull request. Will remove

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java (hunk 4 above)
Comment (prinsniels) [reply app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java:12]: In the near future the recorder should also start the correct service (pre or post API LVL 26). I picked the name in advance

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java (hunk 3 above)
Comment (prinsniels) [reply app/src/main/java/cellscanner/wowtor/github/com/cellscanner/recorder/Recorder.java]: OK

File: app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java (hunk 2 above)
Comment (prinsniels) [reply app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java]: Check is removed. The will result in unwanted behaviour when not merged with "feature/stop_thread_after_stop_recording"

**Pull request merged**